### PR TITLE
[Refactor][CONV1D] Align convolution padding semantics

### LINKED ISSUE #1025: [REFACTOR][CONV1D] align op api with manifest padding semantics
### Symptom / Motivation

Part of #853.

`Conv1dFwdOp` and `Conv1dBiasFwdOp` are still marked `spec-only` because the implementation does not fully match the manifest-declared `torch.nn.functional.conv1d` API. This issue extracts the low-risk op-layer portion from #853 so it can be reviewed independently before the kernel changes for `dilation` and `groups`.

The current `Conv1dFwdOp.forward()` signature uses `x` for the input tensor, while the manifest and PyTorch API use `input`. The op constructor also accepts only integer `padding`, while the manifest allows 1-element tuple/list forms and string padding values `"valid"` and `"same"`.

### Root Cause Analysis

The Conv1d op layer in `tileops/ops/convolution.py` has not normalized the public constructor/API surface to the manifest contract:

- `Conv1dFwdOp.forward()` takes `x` instead of `input`.
- `stride` and `padding` are typed and passed through as scalar ints only.
- `padding='valid'` is not mapped to `0` before kernel dispatch.
- `padding='same'` is not accepted or validated at the op layer.

This creates a manifest/API gap even before the remaining kernel work for `dilation` and `groups`.

### Related Files

- `tileops/ops/convolution.py`
- `tests/ops/test_convolution.py`
- `tileops/ops_manifest.yaml`

## Goal

Align the Conv1d op-layer API with the manifest-declared PyTorch-compatible surface for parameter naming and padding normalization, without changing kernel math or manifest status.

## Plan

1. Rename the first argument of `Conv1dFwdOp.forward()` from `x` to `input` and pass it through unchanged to the kernel.
2. Add op-layer normalization for 1-element tuple/list values for `stride` and  `padding`.
3. Add support for `padding='valid'` by resolving it to integer padding `0` before kernel construction.
4. Add support for `padding='same'` for the PyTorch-compatible case `stride == 1`, resolving it to the integer padding required by the current kernel path.
5. Raise a clear `ValueError` for unsupported padding strings and for `padding='same'` with `stride != 1`.
6. Add focused tests for the normalized API behavior and keep kernel behavior unchanged for existing scalar-int cases.

## Constraints

- Do not implement `dilation` in this issue.
- Do not implement `groups` in this issue.
- Do not modify `tileops/ops_manifest.yaml` status in this issue.
- Keep changes scoped to op-layer API normalization and focused tests.

## Acceptance Criteria

- [ ] `Conv1dFwdOp.forward()` uses `input` as the first tensor parameter name.
- [ ] `Conv1dFwdOp` accepts 1-element tuple/list values for `stride` and `padding`.
- [ ] `Conv1dFwdOp` accepts `padding='valid'` and produces the same output as integer padding `0`.
- [ ] `Conv1dFwdOp` accepts `padding='same'` when `stride == 1` and matches `torch.nn.functional.conv1d` for covered odd-kernel cases.
- [ ] Invalid padding strings raise `ValueError` with a clear message.
- [ ] `padding='same'` with `stride != 1` raises `ValueError`.
- [ ] Existing Conv1d scalar-int tests continue to pass via `pytest tests/ops/test_convolution.py -vvs`.

### PR DESCRIPTION
**Pull request by RMLYC**
Closes #1025

## Summary

- Generalize convolution tuple and padding normalization across Conv1d, Conv2d, and Conv3d.
- Add op-layer support for padding="valid" and symmetric odd-kernel padding="same" without changing kernel math or manifest status.
- Rename Conv1d forward input parameter from x to input to match the manifest API.

## Test plan

- [x] pre-commit hooks passed during commit
- [x] python -m ruff check tileops/ops/convolution.py tests/ops/test_convolution.py
- [x] python -m pytest tests/ops/test_convolution.py -q

## Regression

- Existing convolution test coverage remains stable: 24 passed, 8 skipped.
- No test files or manifest status were changed in this PR.

### CODE REVIEW

File: tileops/ops/convolution.py [hunk 1]
```diff
@@ -30,9 +68,12 @@ def __init__(
...
         self.c_out = c_out
-        self.kernel_size = kernel_size
-        self.stride = stride
-        self.padding = padding
+        kernel_size_tuple = _conv_tuple(kernel_size, 1, "kernel_size", "Conv1d")
+        stride_tuple = _conv_tuple(stride, 1, "stride", "Conv1d")
+        padding_tuple = _conv_padding_to_tuple(padding, stride_tuple, kernel_size_tuple, "Conv1d")
+        self.kernel_size = kernel_size_tuple[0]
+        self.stride = stride_tuple[0]
+        self.padding = padding_tuple[0]
```
[the remark this answers is not in the document]
Comment (RMLYC) [reply tileops/ops/convolution.py:146]: Thanks for the review. I kept `Conv1dFwdOp.kernel_size`, `stride`, and `padding` as scalar ints intentionally because the Conv1d kernel constructor consumes scalar `kernel_l`, `stride_l`, and `pad_l`, unlike the 2D/3D paths that naturally index tuple attributes per axis. Storing scalar attributes here keeps the op attributes aligned with the current Conv1d kernel contract.

I did take the type-hint part of the feedback: commit `229bd12` updates `kernel_size` to accept `int | Tuple[int]`. Per follow-up scope, the public accepted structured form is tuple-only, so list support was removed from `kernel_size`, `stride`, and `padding`.

File: tileops/ops/convolution.py [hunk 2]
```diff
@@ -44,9 +85,9 @@ def __init__(
             c_in=c_in,
             l_in=l_in,
             c_out=c_out,
-            kernel_l=kernel_size,
-            stride_l=stride,
-            pad_l=padding,
+            kernel_l=self.kernel_size,
+            stride_l=self.stride,
+            pad_l=self.padding,
```
[the remark this answers is not in the document]
Comment (RMLYC) [reply tileops/ops/convolution.py:160]: I am leaving the Conv1d kernel initialization as scalar access in commit `229bd12`: `kernel_l=self.kernel_size`, `stride_l=self.stride`, and `pad_l=self.padding`. Conv1d stores these attributes as ints because the kernel constructor expects int values, so indexing with `[0]` would only be needed if we changed the attribute storage model to tuples. That tuple storage change was intentionally not adopted for Conv1d.

**Review by Ibuki-wind — APPROVED** [review]
